### ISSUE TITLE
Self-closing REACT components in translation strings should not attempt to replace the component's children

### ISSUE THREAD

**Issue by simkentys** [i18next/react-i18next#1815]
## 🐛 Bug Report

When using React components with children to replace self closing tags in translations, the children of react component is replaced.
Probably continuation of this issue: https://github.com/i18next/react-i18next/issues/1695

## To Reproduce

https://stackblitz.com/edit/vitejs-vite-f6qrql?file=package.json,src%2Fmain.tsx%3AL41&terminal=dev

```js
i18n.use(initReactI18next).init({
  interpolation: {
    escapeValue: false,
  },
});

function Link({ href, children }) {
  return <a href={href}>{children}</a>;
}

createRoot(document.getElementById('root')!).render(
  <>
      <Link href="#">linkTo</Link>
      <Trans
        i18nKey="<Component />"
        components={{ Component: <Link href="#">linkTo</Link> }}
      />
      <Trans
        i18nKey="<Component />"
        components={{ Component: <a href="#">linkTo</a> }}
      />
  </>
);
```

expected result:
```html
<a href="#">linkTo</a>
<a href="#">linkTo</a>
<a href="#">linkTo</a>
```
actual result:
```html
<a href="#">linkTo</a>
<a href="#"></a>
<a href="#">linkTo</a>
```

## Expected behavior

React component children would be preserved

## Your Environment

"i18next": "^24.0.2",
"react-i18next": "^15.1.2"

**Comment by adrai** [comment]
Feel free to provide a PR to address this.

**Comment by adrai** [comment]
addressed in v15.1.3
